src/ddspdrum/torchmodule.py

```diff
@@ -486,12 +529,164 @@ def _forward(self, control_in: T, audio_in: T) -> T:
         assert (control_in >= 0).all() and (control_in <= 1).all()
 
         if (audio_in <= -1).any() or (audio_in >= 1).any():
-            normalize(audio_in)
+            util.normalize(audio_in)
 
-        audio_in = fix_length(audio_in, len(control_in))
+        audio_in = util.fix_length(audio_in, len(control_in))
         return control_in * audio_in
 
 
+class TorchNoise(TorchSynthModule):
+    """
+    Adds noise to a signal
+
+    Parameters
+    ----------
+    ratio (float): mix ratio between the incoming signal and the produced noise
+    **kwargs: see TorchSynthModule
+    """
+
+    def __init__(
+            self,
+            ratio: float = 0.25,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.add_parameters(
+            [
+                TorchParameter(
+                    value=ratio,
+                    parameter_name="ratio",
+                    parameter_range=ParameterRange()
+                )
+            ]
+        )
+
+    def _forward(self, audio_in: T) -> T:
+        noise = self.noise_of_length(audio_in)
+        return util.crossfade(audio_in, noise, self.p("ratio"))
+
+    @staticmethod
+    def noise_of_length(audio_in: T) -> T:
+        return torch.rand_like(audio_in) * 2 - 1
```

Comment:
We'll fix this later: https://github.com/turian/torchsynth/issues/76

Reply:
This creates the random values on the same device as audio_in, so if audio_in is on the GPU then this is equivalent to torch.rand_like(audio_in, device="cuda")